ouroboros-consensus-cardano/tools/db-analyser/Main.hs

```diff
@@ -0,0 +1,327 @@
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications    #-}
+
+module Main (main) where
+
+import           Control.Monad.Except
+import qualified Data.ByteString as BS
+import           Data.Either (fromRight)
+import           Data.Foldable (asum)
+import           Data.IORef
+import           Data.List (intercalate)
+import           Data.Proxy (Proxy (..))
+import qualified Data.Text as Text
+import           GHC.Natural (Natural)
+import           Options.Applicative
+import           System.FilePath.Posix ((</>))
+
+import           Cardano.Slotting.Slot
+
+import           Cardano.Crypto.ProtocolMagic
+
+import           Ouroboros.Network.Block (HasHeader (..), HeaderHash,
+                     genesisPoint)
+
+import           Ouroboros.Consensus.Block
+import           Ouroboros.Consensus.Config
+import           Ouroboros.Consensus.Node.DbMarker
+import           Ouroboros.Consensus.Node.Run
+import           Ouroboros.Consensus.Util.ResourceRegistry
+
+import           Ouroboros.Consensus.Storage.ChainDB.API (BlockComponent (..),
+                     StreamFrom (..), StreamTo (..))
+import           Ouroboros.Consensus.Storage.ChainDB.Impl.ImmDB hiding
+                     (withImmDB)
```

Comment:
I'm now wondering what you're importing here. Depending on the answer, it might make more sense to just import what you need instead of hiding `withImmDB`.

Reply:
I think there are many things imported from the ImmDB: stream, defaultArgs, Iterator, IteratorResult, ChunkInfo, ValidationPolicy, getPointAtTip, ImmDB, ImmDbArgs (..) to name a few

Reply:
Ah, ok :+1: